src/Perscom/Repository/AwardNominationRepository.php

```diff
@@ -0,0 +1,115 @@
+    }
+
+    /** @return AwardNominationData[] $ */
+    public function getAwardNominations(QueryBuilder $qb): Array {
+        $dbResult = $qb->getQuery()->getArrayResult();
+        
+        $perscom = $this->perscomFactory->getPerscom();
+        $this->statusesCache = $perscom->statuses()->all()->json('data');
+        $this->awardsCache = $perscom->awards()->all(include: ['image'])->json('data');
+
+        $rows = [];
+        foreach ($dbResult as $row) {
+            $nomination = AwardNomination::fromArray($row);
+            $data = $this->resolveData($nomination);
+            array_push($rows, $data);
+        }
+
+        return $rows;
+    }
+
+    private function resolveData(AwardNomination $nomination): AwardNominationData
+    {
+        $data = new AwardNominationData();
+        $data->nomination = $nomination;
+        $data->requesterItem = $this->getPerscomUser($nomination->getPerscomUserId());
+        $data->receiverItem = $this->getPerscomUser($nomination->getReceiverUserId());
+
+        if ($nomination->getUpdatedByUserId()) {
+            $data->updatedByItem = $this->getPerscomUser($nomination->getUpdatedByUserId());
+        }
+
+        $data->awardItem = array_find($this->awardsCache, function($x) use ($nomination) { return $x['id'] == $nomination->getAwardId(); });
+        
+        $status = array_find($this->statusesCache, function ($x) use ($nomination) {
+            return $x['id'] == $nomination->getStatus();
+        });
+
+        $data->statusItem = $status;
+
+        return $data;
+    }
+
+    public function getAwardNomination(int $id): ?AwardNominationData
+    {
+        $qb = $this->getListQuery()
+        ->andWhere('ar.id = :id')
+        ->setParameter('id', $id);
+
+        $nominations = $this->getAwardNominations($qb);
+        if (count($nominations) > 0) {
+            return $nominations[0];
+        }
+
+        return null;
+    }
+
+    // Because of limit in perscom api, we need to be able to navigate all the pages until there are no more
+    private function getPerscomUser(int $id, ?int $lastPage = null): ?Array {
+        // page in perscom api is 1-based
+        $lastLocalPage = count($this->usersCache) + 1;
+        foreach ($this->usersCache as $pageData) {
+            $user = array_find($pageData, function($x) use ($id) { return $x['id'] == $id; });
+            if ($user != null)
+                return $user;
+        }
+
+        // If we are at last page we don't wanna query another time
+        if ($lastLocalPage == $lastPage) {
+            return null;
+        }
+
+        $perscom = $this->perscomFactory->getPerscom();
+        $result = $perscom->users()->all(page: $lastLocalPage, limit: 100)->json();
+        $meta = $result['meta'];
+        $lastPage = $meta['last_page'];
+        $data = $result['data'];
+        $this->usersCache[$lastLocalPage] = $data;
+
+        return $this->getPerscomUser($id, $lastPage);
+    }
```

Comment:
3rdid had 5000 p-files on invision. On here that would mean (worst case scenario) 50 calls to the perscom api....

That'd take longer than the request timeout of 30 seconds most standard webservers have.